keyring/secrets.go

```diff
@@ -0,0 +1,239 @@
+// +build linux
+
+package keyring
+
+import (
+  "github.com/gsterjov/go-libsecret"
+  "fmt"
+  "os"
+  "encoding/json"
+)
+
+const SECRETS_COLLECTION = "awsvault"
+
+
+func init() {
+  service, err := libsecret.NewService()
```

Comment:
This should be moved into the opener below, otherwise it will result in errors on any system without libsecret.